### ISSUE TITLE
parse_config_file resource fails when an empty file is provided

### ISSUE THREAD

**Issue by rx294** [inspec/inspec#2805]
### 🎛 Description

parse_config_file fails when an empty file is provided.

Expected behavior:

resource should skip if file is non existent or unreadable.
if the file is empty no params should be returned.  

### 🌍 InSpec and Platform Version

inspec-2.0.45

### 🤔 Replication Case

The bug appears when an empty file is provided to parse_config_file
the code bug resides in line 78

```
77     if @content.empty? && !inspec.file(conf_path).empty?
78         return skip_resource "Can't read file \"#{conf_path}\""
79     end`
```

file resource does not have an `empty?` feature.

### 💁 Possible Solutions

Current Code:

```
    def parse_file(conf_path)
      @conf_path = conf_path

      # read the file
      if !inspec.file(conf_path).file?
        return skip_resource "Can't find file \"#{conf_path}\""
      end
      @content = read_file(conf_path).to_s
      if @content.empty? && !inspec.file(conf_path).empty?
        return skip_resource "Can't read file \"#{conf_path}\""
      end

      read_params
    end

```

Fix Code:
```
    def parse_file(conf_path)
      @conf_path = conf_path

      # read the file
      if !inspec.file(conf_path).file?
        return skip_resource "Can't find file \"#{conf_path}\""
      end
      @content = read_file(conf_path).to_s
      if @content.nil?
        return skip_resource "Can't read file \"#{conf_path}\""
      end

      read_params
    end
```

the above code will skip the resource if file is non-existent or unreadable.

### 💻 Stacktrace

```
inspec> parse_config_file('empty.conf')
NoMethodError: undefined method `empty?' for File empty.conf:#<Class:0x00007f98d4aa19e0>
from /Users/libyxavier/.rvm/gems/ruby-2.5.0/gems/inspec-2.0.45/lib/resources/parse_config.rb:78:in `parse_file'
```

@aaronlippold @samcornwell

**Comment by jquick** [comment]
@rx294 - Thanks for reporting this!

**Comment by TrevorBramble** [comment]
Fixed by #2359
